text/0059-token-metadata-document.md

```diff
@@ -0,0 +1,485 @@
+in an appropriate way.
+
+This proposal differs from an earlier proposal MCIP 40, in that it is a stop-gap mechanism which
+does not post metadata to the blockchain.
+
+This proposal specifies how clients should dynamically discover token metadata, authenticate this data,
+and how they can go on to use this data.
+
+# Motivation
+[motivation]: #motivation
+
+When a client gets a `TxOut`, at the lowest level, both the `value` and `token_id` are `u64`
+numbers which can be decrypted from the blockchain records. However, it isn't appropriate to
+display the `token_id` number directly to the users, since they won't know what this means.
+
+Metadata about a `token_id` consists of all the data which helps clients figure out how they
+can appropriately represent value to the users. Tampering with this metadata can represent an attack
+to confuse or steal from the users, so there is considerable interest in ensuring that there
+is a robust way for clients to authenticate the metadata.
+
+If we do not support dynamic discovery of token metadata, then it must always be baked into clients,
+which is what we do today.
+
+Because some clients have relatively long upgrade times, this means that new
+tokens that are launched still cannot actually be used by users. This effectively hurts the
+time-to-market on any new token that is created.
+
+There are second order effects to this -- if tokens become "available" in clients at different
+points in time, then users can get bitten in a bad way. For instance, if a Moby user sent eUSD
+to a Signal user, and at the time, Moby supports eUSD but Signal does not, then the recipient
+cannot see the payment in their app, or send it anywhere, or send it back to the Moby user.
+This could lead to disputed payments, bad user experiences, etc.
+
+At the same time, putting token metadata on chain will require an enclave upgrade, while creating
+new tokens on the chain does not. So there is considerable interest in having an off-chain source
+of token metadata that can be validated by clients.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+This proposal creates a source of truth for token metadata for tokens on the MobileCoin blockchain.
+
+Two documents are hosted on the MobileCoin foundation website:
+
+* `https://www.mobilecoin.com/token_metadata.json`: A JSON document containing token metadata
+* `https://www.mobilecoin.com/token_metadata.sig`: An ed25519 signature over the bytes of the .json document. This is binary data.
+
+The ed25519 signing key will be controlled by the mobilecoin foundation.
+This MCIP will be updated, and the public key will be posted here, after the key is created,
+if the proposal is accepted.
+
+Token metadata ed25519 public key hex bytes: `TBD`
+
+The `token_metadata.json` document has the following schema (example):
+
+```
+{
+    signing_timestamp: "10020032320032",
```

Comment:
Maybe we should include a schema_version field that is initially set to 1, and incremented when we make changes to the schema/start including other data/etc

Reply:
good idea, we can add that too